openedx/core/djangoapps/content/block_structure/block_structure.py

```diff
@@ -466,6 +466,26 @@ def get_xblock_field(self, usage_key, field_name, default=None):
         block_data = self._block_data_map.get(usage_key)
         return getattr(block_data, field_name, default) if block_data else default
 
+    def override_xblock_field(self, usage_key, field_name, override_data):
+        """
+        Set value of the xBlock field for the requested block for the requested field_name;
+
+        Arguments:
+            usage_key (UsageKey) - Usage key of the block whose xBlock
+                field is requested.
+
+            field_name (string) - The name of the field that is
+                requested.
+
+            override_data (object) - The data you want to set
+        """
+        block_data = self._block_data_map.get(usage_key)
+        setattr(
+            block_data,
+            field_name,
+            override_data
+        )
```

Comment:
Do we have to worry about inherited values?

Reply:
what should be the priory inherited values or overridden values?

Reply:
I don't know. Sorry, I wasn't being clear. The main thing I'm concerned about is that whatever we're doing here is consistent with courseware in terms of how it handles inherited values for IDDE (if that's even relevant).

Reply:
As far as I know it is consistent, but I want an expert opinion from edX.

Reply:
Field overrides and IDDE are MIT contributions, so the experts on those areas are likely all at MIT ODL (@pdpinch might be able to tag someone?). I think I'm more familiar with it than most at edX, but I'd still end up just making a test case or checking manually after setting up some data. Sorry.

Reply:
Due date extensions are only applied at the vertical, which is the only place that due dates are set. So I don't think we need any inheritance. 

It is the case that this is a more generic implementation, so I'm not sure if some future use of a student override might need inheritance. Can we cross that bridge when we come to it?

Reply:
@pdpinch this due date is applied to subsection/sequential and from it downward i.e verticals and problem. So subsection are the  starting point. you can test it by change due date in studio for a subsection and the inside lms go to extension and override that date and the change check the courseware and progress page (the represent subsection)

Reply:
Sorry, I misspoke. You are correct that the due date is applied at the sequential and I've confirmed that it is inherited as expected.